core/src/com/unciv/ui/screens/worldscreen/WorldMapHolder.kt

```diff
@@ -695,7 +695,8 @@ class WorldMapHolder(
             val attackableTiles: List<AttackableTile> =
                 if (nukeBlastRadius >= 0)
                     selectedTile!!.getTilesInDistance(nukeBlastRadius)
-                        .filter { it.getFirstUnit() != null }
+                        // Should not display invisible submarine units even if the tile is visible.
+                        .filter { targetTile -> (targetTile.isVisible(unit.civ) && targetTile.getUnits().any { !it.isInvisible(unit.civ) }) || (targetTile.isCityCenter() && unit.civ.hasExplored(targetTile)) }
```

Comment:
personally I think this line would look way more readable if each condition was on it's own line